python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -0,0 +1,518 @@
+                # Scale the noise by the square root of the variance.
+                var.sqrt()  # in-place square root
+                noise *= var
+
+            # 4. Add the scaled noise to the PSF image.
+            psfImage += noise
+
+            # Masking is needed for debiased PSF moments.
+            badpix = exposure[bbox].mask.array.copy()
+            bitValue = exposure.mask.getPlaneBitMask(badMaskPlanes)
+            badpix &= bitValue
+
+            # Convert the mask of bad pixels to a format suitable for galsim.
+            # NOTE: galsim.Image will match whatever dtype the input array is
+            # (here int32).
+            badpix = galsim.Image(badpix, bounds=bounds)
+        else:
+            badpix = None
+
+        # Extract the numpy array underlying the PSF image.
+        imageArray = psfImage.array
+
+        # Create a GalSim image using the PSF image array.
+        image = galsim.Image(imageArray, bounds=bounds, copy=False)
+
+        # Decide on the centroid position based on configuration.
+        if self.config.useSourceCentroidOffset:
+            # If the source centroid offset should be used, use the source
+            # centroid.
+            centroid = center
+        else:
+            # Otherwise, use the center of the center of the bounding box of
+            # psfImage
+            centroid = Point2D(psfImageBBoxAtCenter.getMin() + psfImageBBoxAtCenter.getDimensions() // 2)
+
+        # Call the internal method to calculate adaptive moments using GalSim.
+        self._calculate(
+            record,
+            image=image,
+            weight=None,
+            badpix=badpix,
+            sigma=psfSigma,
+            centroid=centroid,
+        )
+
+    def measure(self, record, exposure):
+        self._measure(record, exposure, debiasedPsfMoments=False)
+
+
+class HsmPsfMomentsDebiasedConfig(HsmPsfMomentsConfig):
+    """Configuration for debiased HSM adaptive moments measurement for PSFs."""
+
+    noiseSource = pexConfig.ChoiceField[str](
+        doc="Noise source. How to choose variance of the zero-mean Gaussian noise added to image.",
+        allowed={
+            "meta": "variance = the 'BGMEAN' metadata entry",
+            "variance": "variance = the image's variance plane",
+        },
+        default="variance",
+    )
+    seedOffset = pexConfig.Field[int](doc="Seed offset for random number generator.", default=0)
+    badMaskPlanes = pexConfig.ListField[str](
+        doc="Mask planes used to reject bad pixels.", default=["BAD", "SAT"]
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.useSourceCentroidOffset = True
+
+
+@measBase.register("ext_shapeHSM_HsmPsfMomentsDebiased")
+class HsmPsfMomentsDebiasedPlugin(HsmPsfMomentsPlugin):
+    """Plugin for debiased HSM adaptive moments measurement for PSFs."""
+
+    ConfigClass = HsmPsfMomentsDebiasedConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER + 1
```

Comment:
```suggestion
        return cls.FLUX_ORDER + 0.1
```
Since the standard execution order increases in steps of 1, it's safer to keep the increase by hand to less than 1. The exact value does not matter.